src/weights.jl

```diff
@@ -682,6 +682,31 @@ function mean(A::AbstractArray, w::UnitWeights; dims::Union{Colon,Int}=:)
     return mean(A, dims=dims)
 end
 
+"""
+    mean(f, A::AbstractArray, w::AbstractWeights[, dims::Int])
+
+Compute the weighted mean of array `A`, after transforming it'S
+contents with the function `f`, with weight vector `w` (of type
+`AbstractWeights`). If `dim` is provided, compute the
+weighted mean along dimension `dims`.
+
+# Examples
+```julia
+n = 20
+x = rand(n)
+w = rand(n)
+mean(√, x, weights(w))
+```
+"""
+mean(f, A::AbstractArray, w::AbstractWeights; dims::Union{Colon,Int}=:) =
+    _mean(f.(A), w, dims)
```

Comment:
Avoid memory allocation by using `mean(f, A)` instead of `mean(f.(A))`. Remember that `f.(A)` creates an extra array, which is slow. Memory access is usually the biggest bottleneck on modern CPUs. `mean(f.(A))` is 2 separate operations: The first one creates a new array, `f.(A)`, and the second calculates its mean. `mean(f, A)` calculates the mean of `(f(x) for x in A)` directly, as one operation, without creating a new array.
```suggestion
    _mean(f, A; dims)
```
I'd also suggest making it slightly more generic, as 
```suggestion
    _mean(f, A; kwargs...)
```
(See below for more details.)